homeassistant/components/sun/__init__.py

```diff
@@ -95,6 +96,9 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up from a config entry."""
+    # Process integration platforms right away since
+    # we will create entities before firing EVENT_COMPONENT_LOADED
+    await async_process_integration_platform(hass, DOMAIN)
```

Comment:
I am confused why this is needed. We don't have a `sun.py` platform.

Reply:
Maybe I should rename `async_process_integration_platform` to `async_process_integration_platform_component`

Its a bit confusing

Reply:
https://github.com/home-assistant/core/pull/70217